src/styles/stars/stars.scss

```diff
@@ -0,0 +1,25 @@
+.stars {
+  display: flex;
+}
+
+.stars__star {
+  background-image: url(../images/star.svg);
+  background-position: center;
+  background-repeat: no-repeat;
+  width: 16px;
+  height: 16px;
+  margin-right: 2px;
+  border-radius: 0.5px;
+}
+
+.stars__star:not(:first-child) {
+  margin-left: 2px;
+}
+
+.stars--5 > .stars__star,
+.stars--1 > .stars__star:first-child,
+.stars--2 > .stars__star:nth-child(-1n + 2),
+.stars--3 > .stars__star:nth-child(-1n + 3),
+.stars--4 > .stars__star:not(:last-child) {
```

Comment:
Try to optimize you solution with [loop](https://sass-lang.com/documentation/at-rules/control/for)